pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -241,16 +266,124 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) (ctrl.Result, erro
 		}
 	}
 
-	// Knative does not support INIT containers or mounting, so we add annotations that trigger the
-	// StorageInitializer injector to mutate the underlying deployment to provision model data
-	if sourceURI := predictor.GetStorageUri(); sourceURI != nil {
-		if _, ok := annotations[constants.StorageInitializerSourceUriInternalAnnotationKey]; ok {
-			return ctrl.Result{}, errors.New("must provide only one of storageUri and storage.path")
+	if multiNodeEnabled {
```

Comment:
Can we move this big block into a separate function ? this method is becoming unmanageable and hard to review

Reply:
also the container merging logic maybe can be generalized so we do not repeat the similar code

Reply:
The block of code has been extracted into a separate function called `multiNodeProcess`. Additionally, the container merging logic has been generalized and moved to `MergeServingRuntimeAndInferenceServiceSpecs` functions in utils.go file for shared use.